neo4j-plugins/src/main/java/io/cdap/plugin/transformer/MdmLookupTransformer.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2015-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.transformer;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.InvalidEntry;
+import io.cdap.cdap.etl.api.Transform;
+import io.cdap.cdap.etl.api.TransformContext;
+import io.cdap.plugin.sink.Neo4jDataService;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * MdmLookup plugin to verify if data present in MDM.
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name("MdmLookup")
+@Description("Verify whether the requested value is present in MDM.")
+public class MdmLookupTransformer extends Transform<StructuredRecord, StructuredRecord> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MdmLookupTransformer.class);
+
+    private final LookupConfig config;
+    private Neo4jDataService dataService;
+    private String lookupColumn;
+    private Driver driver;
+
+    public MdmLookupTransformer(LookupConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) {
+        List<String> lookupData = new ArrayList<>();
+        for (Schema.Field field : input.getSchema().getFields()) {
+            processField(input, lookupData, field, false);
+        }
+
+        final List<String> notFoundList = lookupData.stream()
+            .filter(uuid -> dataService.getUniqueNodeByProperty(lookupColumn, uuid) == null)
+            .collect(Collectors.toList());
+
+        if (!notFoundList.isEmpty()) {
+            notFoundList.forEach(id -> emitter.emitError(
+                new InvalidEntry<>(1, String.format("Not found in MDM. UUID: %s", id), input)));
```

Comment:
The error code needs to be documented appropriately so the message doesn't need to be present in response. The message filed just needs to contain the UUID that was not found.

Reply:
now message contains only UUID that was not found